platforms/opencl/src/OpenCLContext.cpp

```diff
@@ -286,9 +293,21 @@ OpenCLContext::OpenCLContext(const System& system, int platformIndex, int device
             compilationDefines["SYNC_WARPS"] = "barrier(CLK_LOCAL_MEM_FENCE)";
         vector<cl::Device> contextDevices;
         contextDevices.push_back(device);
-        cl_context_properties cprops[] = {CL_CONTEXT_PLATFORM, (cl_context_properties) platforms[bestPlatform](), 0};
+        vector<cl_context_properties> cprops = {CL_CONTEXT_PLATFORM, (cl_context_properties) platforms[bestPlatform]()};
+        if (platforms[bestPlatform].getInfo<CL_PLATFORM_EXTENSIONS>().find("cl_khr_throttle_hints") != string::npos) {
+            cprops.push_back(CL_QUEUE_THROTTLE_KHR);
```

Comment:
@peastman Here, I think you are mixing the queue properties with context properties!
I suggest st like below,
```cpp
        if (platforms[bestPlatform].getInfo<CL_PLATFORM_EXTENSIONS>().find("cl_khr_throttle_hints") != string::npos) {
            vector<cl_queue_properties> qprops = {CL_QUEUE_THROTTLE_KHR};
            if (throttle == "low")
                qprops.push_back(CL_QUEUE_THROTTLE_LOW_KHR);
            else if (throttle == "medium")
                qprops.push_back(CL_QUEUE_THROTTLE_MED_KHR);
            else if (throttle == "high")
                qprops.push_back(CL_QUEUE_THROTTLE_HIGH_KHR);
            else
                throw OpenMMException("Illegal value for OpenCLThrottle");
            qprops.push_back(0);
        }
```
and then you need `clCreateCommandQueueWithProperties`, but I think you should update the `CommandQueue` interface in the `opencl.hpp` file, since currently it seems to accept only `QueueProperties`.